dask/dataframe/tests/test_utils_dataframe.py

```diff
@@ -424,3 +426,12 @@ def func():
 
     with pytest.raises(ValueError, match=re.escape("Missing: ['D']")):
         apply_and_enforce(_func=func, _meta=meta)
+
+
+@pytest.mark.parametrize("cols", ["A", ["A"], ["A", "B"]])
+def test_drop_by_shallow_copy(cols):
+
+    df = pd.DataFrame(np.random.randn(10, 3), columns=list("ABC"))
+    df2 = drop_by_shallow_copy(df, cols)
+
+    assert_eq(df2, df.drop(columns=cols))
```

Comment:
I think that we probably don't need to test this directly.  The behavior of drop should already be tested at the public API level.  I don't think that it's necessary to test this internal operation.  

What might be more important to test is that we don't perform a deep copy, so maybe checking that `df2.C._values is df.C._values`?

Reply:
Looks like we have a problem here (https://github.com/pandas-dev/pandas/issues/17406).  I was revising the PR and tweaked the test to check that the copy is actually shallow, but the test fails.  Seems that you can successfully shallow copy a series in pandas, but the dataframe shallow copy does not work...

Reply:
Hrm, indeed, but it does still seem to be fast.

```python
In [1]: import pandas as pd, numpy as np

In [2]: df = pd.DataFrame({"x": np.arange(10000000)})

In [3]: %timeit df.copy()
20.3 ms ± 1.22 ms per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [4]: %timeit df.copy(deep=False)
39.5 µs ± 2.38 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [6]: df2 = df.copy(deep=False)

In [7]: df2.x._values is df.x._values
Out[7]: False
```

I suspect that the fact that the output of the last line is False has more to do with my not understanding how the `._values` accessor works than with copying.

Reply:
```python
In [31]: df2.x._values.base is df.x.values.base
Out[31]: True
```

I'd recommend `np.may_share_memory`

```python
In [32]: np.may_share_memory(df2.x.values, df.x.values)
Out[32]: True
```

Reply:
> I suspect that the fact that the output of the last line is False has more to do with my not understanding how the ._values accessor works than with copying.

@TomAugspurger - Do you know the *correct* way to test this?  The [pandas docs](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.copy.html) show an example for series (which works fine for me), but not for dataframes.

Reply:
Hopefully answered in https://github.com/dask/dask/pull/5675#discussion_r354413037, but LMK if not.

Reply:
Regardless, I'm also okay with removing the test if we know the change isn't slowing pandas-based code down, I know it is significantly boosting dask_cudf performance :)

Reply:
Thanks for your help/advice @TomAugspurger!

I have still been struggling with this for longer than I'd like to admit :/ .  It seems that the drop operation is creating a copy in pandas (in certain cases):

```python
In [1]: import pandas as pd; import numpy as np 
   ...: df = pd.DataFrame(np.random.randn(10, 3), columns=list("ABC")) 
   ...: df2 = df.copy(deep=False)                                                                                                                                                                                                

In [2]: np.may_share_memory(df2.C.values.base, df.C.values)                                                                                                                                                                      
Out[2]: True

In [3]: df = pd.DataFrame(np.random.randn(10, 3), columns=list("ABC")) 
   ...: df2 = df.copy(deep=False) 
   ...: del df2["A"] 
   ...: np.may_share_memory(df2.C.values.base, df.C.values)                                                                                                                                                                      
Out[3]: False
```

The really wierd part is that there is NOT a copy made if you explicitly check that the memory is shared **before** the drop operation:

```python
In [4]: df = pd.DataFrame(np.random.randn(10, 3), columns=list("ABC")) 
   ...: df2 = df.copy(deep=False) 
   ...: np.may_share_memory(df2.C.values.base, df.C.values)                                                                                                                                                                      
Out[4]: True

In [5]: del df2["A"] 
   ...: np.may_share_memory(df2.C.values.base, df.C.values)                                                                                                                                                                      
Out[5]: True
```

Does this behavior make any sense to you?

Reply:
No, that doesn't make sense to me, sorry :)

FWIW, I think at this point we can give up on this test assertion, if it's proving to be too much of a hassle.